### PR TITLE
Device info feedback

### PR DESCRIPTION
**Pull request by btkelly**

### CODE REVIEW

File: amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java [hunk 1]
```diff
@@ -35,6 +37,23 @@ public ApplicationInfoProvider(@NonNull final Context context) {
 
     @NonNull
     @Override
+    public String getDeviceName() {
+        String manufacturer = Build.MANUFACTURER;
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: `final`

File: amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java [hunk 2]
```diff
@@ -35,6 +37,23 @@ public ApplicationInfoProvider(@NonNull final Context context) {
...
+        String model = Build.MODEL;
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: `final`

File: amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java [hunk 3]
```diff
@@ -35,6 +37,23 @@ public ApplicationInfoProvider(@NonNull final Context context) {
...
+
+        String deviceName;
+
+        if (model.startsWith(manufacturer)) {
+            deviceName = StringUtils.capitalize(model);
+        } else {
+            deviceName = StringUtils.capitalize(manufacturer) + " " + model;
+        }
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: I see there's also this:

```
    /** The name of the overall product. */
    public static final String PRODUCT = getString("ro.product.name");
```

which might handle these cases for us?

File: amplify/src/main/java/com/github/stkent/amplify/utils/StringUtils.java [hunk 4]
```diff
@@ -40,6 +40,34 @@ private StringUtils() {
...
+        }
+
+        int strLen = string.length();
+
+        char[] newChars = new char[strLen];
+        newChars[0] = newChar;
+        string.getChars(1, strLen, newChars, 1);
+
+        return String.valueOf(newChars);
+    }
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/utils/StringUtils.java]: Sweet; hows about some tests :) Should be easy enough to mimic the existing string utils tests.

File: amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java (hunk 3 above)
Comment (btkelly) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: No idea, what is the product name exactly?

File: amplify/src/main/java/com/github/stkent/amplify/utils/StringUtils.java (hunk 4 above)
Comment (btkelly) [inline amplify/src/main/java/com/github/stkent/amplify/utils/StringUtils.java]: Lol I knew that would be a comment, thought you might not notice

File: amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java (hunk 3 above)
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: No idea! Might be good to log all these properties in JJs real quick, and see what each outputs on a couple devices?

File: amplify/src/main/java/com/github/stkent/amplify/utils/StringUtils.java (hunk 4 above)
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/utils/StringUtils.java]: :flashlight: :toilet: :hankey: :disappointed: :trollface:

File: amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java (hunk 3 above)
Comment (btkelly) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: MotoX 1st Gen Readings
11-30 14:19:51.484 26127-26127/com.jimmyjohns.debug E/TEST: Product ghost_verizon
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: Board MSM8960
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: BRAND motorola
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: HARDWARE qcom
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: MANUFACTURER motorola
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: MODEL XT1060
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: TAGS release-keys
11-30 14:19:51.485 26127-26127/com.jimmyjohns.debug E/TEST: TYPE user

Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/ApplicationInfoProvider.java]: Then I think the stuff you initially included is the best :)

**Pull request merged**